### ISSUE TITLE
Extend the `define` syntax and scope

### ISSUE THREAD

**Issue by nikivazou** [ucsd-progsys/liquidhaskell#1247]
I want to define `x ==. y = x`, thus `==.` can exist in reflected functions. 

1. Currently, definitions can not be in modules. All definitions should be in should be in https://github.com/ucsd-progsys/liquidhaskell/blob/develop/include/CoreToLogic.lg...

2. Parser does not let me define infix functions (like `==.`)

**Comment by ranjitjhala** [comment]
I'm not sure what the issue is over here? 

Is it that you want to `reflect` the operators 
in the `ProofCombinators` or something else?

Whats an example test that you want to work (that doesn't right now?)

**Comment by nikivazou** [comment]
No, I want to reflect `foo` where `foo` contains `(==.)` in its definition. 

I will push an example soonish.

**Comment by ranjitjhala** [comment]
Yes, that's what I meant too. But for that, you basically
need only to reflect `==.` (because all functions appearing
inside a reflected body must transitively be reflected.)

Right?

**Comment by nikivazou** [comment]
Alternatively, I want to inline "==." in the logic so when reflection sees `x ==. y` is just reflects `y`. 
That is exactly what `define` does. Inlines definitions in the logic!
